src/api/controllers/notificationController.js

```diff
@@ -0,0 +1,170 @@
+      return res.status(status.OK).json(result);
+    } catch (error) {
+      sendError(status.NOT_FOUND, res, 'config', errorMessages.configNotFound);
+    }
+  }
+
+  /**
+   * A method to update user configs
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object containing set configuration
+   */
+  static async updateConfig(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id }
+    } = req.user;
+
+    const isUnsubscribeRequest = req.url.includes('email');
+
+    // Updating the configuration
+    if (isUnsubscribeRequest) {
+      await NotificationConfig.update({
+        config: JSON.stringify(emailUnsubscriptionConfig)
+      },
+      {
+        where: {
+          userId: id
+        }
+      });
+    } else {
+      await NotificationConfig.update({
+        config: JSON.stringify(req.body)
+      },
+      {
+        where: {
+          userId: id
+        }
+      });
+    }
+
+    // sending the result back
+    result.status = status.OK;
+    result.message = 'Configurations updated successfully';
+
+    return res.status(status.CREATED).json({ result });
+  }
+
+  /**
+   * A method to get all notifications of a user
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object containing all notifications
+   */
+  static async getAll(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id }
+    } = req.user;
+
+    const isUnseenRequest = req.url.includes('unseen');
+    let notifications;
+
+    if (isUnseenRequest) {
+      notifications = await Notification.findAll({
+        where: {
+          userId: id,
+          status: 'unseen'
+        }
+      });
+    } else {
+      notifications = await Notification.findAll({
+        where: {
+          userId: id,
+          status: 'seen'
+        }
+      });
+    }
+
+    if (!_.isEmpty(notifications)) {
+      // Sending the result
+      result.status = status.OK;
+      result.data = notifications;
+      return res.status(status.OK).json(result);
+    }
+    sendError(status.NOT_FOUND, res, 'config', errorMessages.notificationNotFound);
+  }
+
+  /**
+   * A method to update a single notification
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object
+   */
+  static async update(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id: userId }
+    } = req.user;
+    const { id } = req.params;
+
+    const isUnseenRequest = req.url.includes('unseen');
+
+    if (isUnseenRequest) {
+      await Notification.update({
+        status: 'unseen'
+      },
+      {
+        where: {
+          id,
+          userId
+        }
+      });
+    } else {
+      await Notification.update({
+        status: 'seen'
+      },
+      {
+        where: {
+          id,
+          userId
+        }
+      });
+    }
+
```

Comment:
I think the method should be changeStatus or something else instead of update only

Reply:
Copy that @alainburindi